### ISSUE TITLE
Supporting for WGBS data ?

### ISSUE THREAD

**Issue by QianzhaoJ** [demh/epigenetic_ageing_clock#1]
Hi :
    I'm very interesting about epigenetic ageing clock. It looks very helpful for many researches.
    Now I had some data using Whole-Genome Bisulfite Sequencing (WGBS), and I wanted to calculate epigenetic ageing clock. But I found these scripts need the 450k array.
    I don't know whether the current algorithm supports WGBS data. Could you give me some suggestions ?
    Thanks in advance !
Qianzhao

**Comment by demh** [comment]
Hi Qianzhao,

Thanks for your query. If you extract the methylation values (as fraction, 0-1, similar to beta-values) for the sequenced cytosines that overlap with the correspondent probes in the 450K array you should be able to obtain a rough estimate of the epigenetic age (e.g. using https://github.com/demh/epigenetic_ageing_clock/blob/master/utils/calculate_DNAmAge.R).

Nevertheless, it is important to keep in mind that the distributions of methylation values are slightly different between WGBS and 450K array, and therefore it is likely that you will observe an offset in the predictions. 

I hope this helps. 

Best wishes,

Dani

**Comment by QianzhaoJ** [comment]
Hi Dani ,
Thanks for your reply !
But l had many questions when I tried to calculate epigenetic ageing clock using `calculate_DNAmAge.R `.  For example , the format of input files ? which data in this repository should I start with?
I have a presumptuous request that I wonder if you can give consent to . I hope that you counld provide a simple manual for estimating of the epigenetic age using data in this repository . 
Anyway, thanks a lot !
Qianzhao

**Comment by demh** [comment]
Hi QianzhaoJ,

Apologies for the super slow reply. I must have missed the email and only saw your response here by chance. 

If you run `Rscript calculate_DNAm_age.R --help` you should get the info that you require. The input for this script is a matrix of beta values (or methylation fraction) for the common 27K/450K probes originally used by Horvath. This matrix can be generated from the following script (using raw IDAT files as input):

https://github.com/demh/epigenetic_ageing_clock/blob/master/utils/idat_to_DNAmAge_methylation_matrix.R

So, in summary:

IDAT files --> idat_to_DNAmAge_methylation_matrix.R --> beta-values matrix --> calculate_DNAm_age.R --> DNAmAge calculations

I will try to add this more explicitly in the documentation. 

In order to adapt this to WGBS you will need to find the common probes in your sequencing data and input that to calculate_DNAm_age.R, but be aware that you may observe some bias/offset between sequencing and array data DNAmAge calculations.

Let me know if this helps. 

Best,

Dani
